tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -3095,110 +3091,206 @@ LogicalResult ConvertTFLFakeQuantOp::matchAndRewrite(
 
 void LegalizeTFL::runOnFunction() {
   OwningRewritePatternList patterns(&getContext());
-  populateLegalizeTFLPatterns(&getContext(), patterns);
+  populateLegalizeTFLPatterns(&getContext(), patterns, legalization_enable);
 
   auto func = getFunction();
   if (ApplyPatternsWithShapeResolution(func, std::move(patterns)).failed()) {
     signalPassFailure();
   }
 }
+
+template <typename TFLOpT, typename TosaOpT>
+class ConvertTFLUnaryOp : public OpConversionPattern<TFLOpT> {
+public:
+  using OpConversionPattern<TFLOpT>::OpConversionPattern;
+  using OpAdaptor = typename TFLOpT::Adaptor;
+  LogicalResult matchAndRewrite(
+    TFLOpT op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+      rewriter.replaceOpWithNewOp<TosaOpT>(
+         op, op.getType(), adaptor.getOperands()[0]);
+      return success();
+  }
+};
+
+template <typename TFLOpT, typename TosaOpT>
+class ConvertTFLBinaryOp : public OpConversionPattern<TFLOpT> {
+public:
+  using OpConversionPattern<TFLOpT>::OpConversionPattern;
+  using OpAdaptor = typename TFLOpT::Adaptor;
+  LogicalResult matchAndRewrite(
+    TFLOpT op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+      rewriter.replaceOpWithNewOp<TosaOpT>(
+         op, op.getType(), adaptor.getOperands()[0], adaptor.getOperands()[1]);
+      return success();
+  }
+};
+
 }  // namespace
 
 void populateLegalizeTFLPatterns(MLIRContext* ctx,
-                                 RewritePatternSet& patterns) {
-  // Add the generated patterns to the list.
+                                 RewritePatternSet& patterns,
+                                 std::unordered_map<std::string, bool> legalization_enable) {
+  #define DEF_PATTERN_INSERT(PAT) patterns.insert<Convert##PAT##Op>(ctx);
```

Comment:
Declare a helper lambda that returns of bool of whether to include. Then perform patterns.insert dependent on that lambda. It is a repeatable function and generates far less binary / code than a macro.

Reply:
After perfroming all the other changes I have ended up with:

```
void populateLegalizeTFLPatterns(MLIRContext* ctx,
                                 RewritePatternSet& patterns,
                                 std::unordered_set<std::string> legalization_disable) {
  //  0. If there is no entry for the Op in the legalization_disable set
  //     than we add the rewrite.
  //  1. If there is an entry for the Op in the legalization_disable set
  //     than we skip adding the rewrite
  #define DEF_PATTERN_INSERT(PAT)                                              \
    if (legalization_disable.find(#PAT) == legalization_disable.end())         \
      patterns.insert<Convert##PAT##Op>(ctx);
```

I think the above seems pretty small so unless we still believe a lambda will better I would prefer to maintain the previous style.